File: eRxWeb/ServerModel/SelectMedicationGridModel.cs
using System.Collections.Generic;
using eRxWeb.ServerModel.Request;
using System.Linq;

namespace eRxWeb.ServerModel
{
    public class SelectMedicationGridModel
    {
        public SelectMedicationGridModel()
        {
            this.Meds = new List<SelectMedicationMedModel>();
        }
        public bool IsAllMedSearch { get { return this.RequestFor == SelectMedicationRequestType.AllMedication; } }
        public List<SelectMedicationMedModel> Meds { get; set; }
        public List<MessageModel> Messages { get; internal set; }
        public SelectMedicationRequestType RequestFor { get; set; }
        public PatientRecordStatus PatientRecordStatus { get; set; }
    }
}

File: eRxWeb/SPA/Model/select-medication/select-medication-grid.model.ts
import { SelectMedicationMedModel, GridPaginationModel, MessageModel, SelectMedicationRequestType, PatientRecordStatus} from '../model.import.def';
export class SelectMedicationGridModel {
    IsPageination: boolean;
    IsAllMedSearch: boolean = false;
    Meds: SelectMedicationMedModel[];
    Messages: MessageModel[];
    RequestFor: SelectMedicationRequestType;
    PatientRecordStatus: PatientRecordStatus;
}

export class SelectMedicationGridPopulationCompletionArgs {
    Messages: MessageModel[];
    RequestFor: SelectMedicationRequestType;
    PatientRecordStatus: PatientRecordStatus;
}

export class SelectMedicationSelections {
    SelectedMeds: SelectMedicationMedModel[];
    EventContext: SelectMedicationEventContext;
}

export class SelectMedicationEventContext {
    CurrentMed: SelectMedicationMedModel;
    CurrentMedIndex: number;
    GridEvent: SelectMedicationGridEvent;
}

export enum SelectMedicationGridEvent {
    MedSelected = 0,
    MedDeselected = 1,
    MedModified = 2,
    AllMedSelected = 3,
    AllMedDeSelected = 4
}